tests/t0116_gateway_cache_test.go

```diff
@@ -310,3 +220,103 @@ func TestGatewayCache(t *testing.T) {
 
 	Run(t, tests)
 }
+
+func TestGatewayCacheWithIPNS(t *testing.T) {
+	fixture := car.MustOpenUnixfsCar("t0116/gateway-cache.car")
+	ipns := ipns.MustOpenIPNSRecordWithKey("t0116/k51qzi5uqu5dlxdsdu5fpuu7h69wu4ohp32iwm9pdt9nq3y5rpn3ln9j12zfhe.ipns-record")
+	ipnsKey := ipns.Key()
+
+	tests := SugarTests{
+		{
+			Name: "GET for /ipns/ unixfs dir listing succeeds",
+			Request: Request().
+				Path("ipns/%s/root2/root3/", ipnsKey),
```

Comment:
unrelated nit:  every time we reference URL path, or IPFS content path, it should start with `/`, 
especially when we use `Equals("/ipns/%s/root2/root3/"` few lines below

Reply:
That's fair, I'll create a follow-up task + PR to fix this!